src/databricks/labs/ucx/source_code/jobs.py

```diff
@@ -139,11 +140,44 @@ def _register_spark_python_task(self, graph: DependencyGraph):
         path = WorkspacePath(self._ws, notebook_path)
         return graph.register_notebook(path)
 
-    def _register_python_wheel_task(self, graph: DependencyGraph):  # pylint: disable=unused-argument
+    @staticmethod
+    def _find_first_matching_distribution(
+        path_lookup: PathLookup, condition: Callable[[metadata.Distribution], bool]
+    ) -> metadata.Distribution | None:
+        for library_root in path_lookup.library_roots:
+            for path in library_root.glob("*.dist-info"):
+                distribution = metadata.Distribution.at(path)
+                if condition(distribution):
+                    return distribution
+        return None
+
+    def _register_python_wheel_task(self, graph: DependencyGraph) -> Iterable[DependencyProblem]:
         if not self._task.python_wheel_task:
-            return
-        # TODO: https://github.com/databrickslabs/ucx/issues/1640
-        yield DependencyProblem('not-yet-implemented', 'Python wheel task is not yet implemented')
+            return []
+
+        def legacy_normalize_name(name: str) -> str:
+            # Prepared exists in importlib.metadata.__init__pyi, but is not defined in importlib.metadata.__init__.py
+            return metadata.Prepared.legacy_normalize(name)  # type: ignore
+
+        distribution_name = self._task.python_wheel_task.package_name
+        distribution = self._find_first_matching_distribution(
+            graph.path_lookup,
+            # Yes, Databricks uses "legacy" normalized name
+            lambda d: legacy_normalize_name(distribution_name) == legacy_normalize_name(d.name),
+        )
```

Comment:
```suggestion
        distribution = self._find_first_matching_distribution(graph.path_lookup, distribution_name)
```
simplify. and use `metadata.Prepared.normalize` inside of `_find_first_matching_distribution`